### ISSUE TITLE
Deadlock on SendFileAsync

### ISSUE THREAD

**Issue by uwx** [NaamloosDT/DSharpPlus#210]
# Summary
Invoking SendFileAsync with the code presented below will cause a deadlock. The file is sent, but the function never returns.

# Details
Windows 7, WS4Net Client, .NETFX 4.7, Lib version 3.2.3

The following NETFX references need to be added to the project:
```
System.Drawing
System.Windows.Forms
System.Windows.Forms.DataVisualization
System.Windows.Forms.DataVisualization.Design
```

This is the code that causes the issue:
```cs
using System.Windows.Forms.DataVisualization.Charting;

        // ...
                _ = Task.Run(async () =>
                {
                    await Task.Yield();
                    
                    var d = new List<DataPoint>();
                    for (var i = 0; i < 10; i++)
                        d.Add(new DataPoint(i, i));
                    using (var s = new MemoryStream())
                    {
                        GeneratePlot(d, s);
                        s.Position = 0;
                        await e.Channel.SendFileAsync(s, "graph.jpg");
                    }
                    Console.WriteLine("ok");
                });
        
        // below
        public static void GeneratePlot(IList<DataPoint> series, Stream outputStream)
        {
            using (var ch = new Chart())
            {
                ch.ChartAreas.Add(new ChartArea());

                var s = new Series {ChartType = SeriesChartType.FastLine};
                foreach (var pnt in series) s.Points.Add(pnt);
                ch.Series.Add(s);
                ch.SaveImage(outputStream, ChartImageFormat.Jpeg);
            }
        }
```

# Steps to reproduce
1. Execute the above code inside the MessageCreated event handler.

# Notes
It seems that the REST bucket thread waits for the heartbeat thread to finish, but of course it never does.

**Comment by Emzi0767** [comment]
Is the file uploaded successfully? i.e. does appear properly?

About the only thing present in that method that *might* cause issues is this, which, to be honest, doesn't look like it could produce issues, assuming you await the returned task.

https://github.com/NaamloosDT/DSharpPlus/blob/release/3.2/DSharpPlus/Entities/DiscordChannel.cs#L169

**Comment by uwx** [comment]
Yes, the file appears perfectly fine

**Comment by Emzi0767** [comment]
Are you able to issue other REST requests after this one "completes"?

**Comment by uwx** [comment]
No.

**Comment by Emzi0767** [comment]
Does the issue occur with any other kind of image supplied?

**Comment by uwx** [comment]
I haven't tested that yet, sorry. I'll do that now

**Comment by uwx** [comment]
fixed in 4.0
